### PR TITLE
Mouse weight form

### PR DESCRIPTION
**Pull request by david-roper**
version 1.0
Closes issue #1 

Form can be seen [here](https://playground.opendatacapture.org/?files=NobwRAxg9gdgLgU3mAXGA9AKgAQIM4A2AlvALQAmReAhgEYELYAOCATgGYIRxGxVzo8UVnFJRaAKy5w82TOgA6MJdBh442ENnIJ2JBAEk1cVgFcAtkg0BfbAF5s1AO7UiGouabC4ACgDk6GbwHgjoAG4AjOgAAlAsMOTUcNQQ1ExwpqyhQTyWpNBZ6CQ6AB4AdBJ4fgCUANwqsOqa2ABe2LYOzq7unt7%2BgabBluFRLVDk0QDMZQBM0yVFCQjllTX1yjDLXiLautSmBBo6eptG6maW8D4gStjYANbFKNh%2BAGIA8gBKALJ%2BADS3bAEagwADmpmooIQzz8SH%2BgOSoLwz2Afm%2BUFMeAQ-xeAHUEERQQALOA4vwAZVSDD8AF0ATA7iREKwYNQCM8bgy7rhKDxYM8IvTudhWZYYd93gBVckAUQA%2BriZQYAOIACQAKnKPj8-IDrELsKpEPAOYC7uYMVj8YSSaaucLHgkYTALLQ2PD7dywtRWEQQXAYSQmKZSQbhcC3ey8QTiRooOxsBbMYwfCRsKDWNRzHhqrrPfqzdg8FSEOS2H6CABhcbQzSFu6O8gw84kUEe4V3b2%2B-3NhAMbjtjsRvvPBRgSlsxhY7sEQ01sdh7lxPlqO0d7ljgAck0mABZJgB2TcABgAbJuxygtzv90ezxewIvhWP3nAiWxL9gX2%2BP2B6%2B1CwLT1i0nMsZ2rHRX3fVg1w7RtRzAcgAE9RSICAF3-HQmGRbBgDHECGDAisIIQMc6X-LIllYHxEmSao609YUiHYGikmoMoCNLcs2RI%2Bw7Acb9oLHapOXXYUsgyFkGLEuCni-MAWzBDDGLErs-RNeSgxDZSZPDOgR3kgAFBhqCxXB4DYbAoB-Vgi242doB0Md-2FawXIAlS7gkzIGRdAgCH-NyVMAxdOIAaTk0TZKdF5FLbJ9sDUntYr7aRBz0yMEIihIrITTidI7ZdeFXaSxLHQzvDoBhPzHclzDZWdtmSehSMfdyx0%2BBBwWBVgarATrup9eTAsAvUDR0ZIiAIHCoruHRi19dJiphABBbB2GEcxsDgKBtszCB7m0Nj1tYKAtqcd9NjCSyQUcGAPDZbAqGwJwY3fcgynS3B1AexByAAEUyJJluwQVCxIc5TG4YqcNRdVdrddapoYcgnoZC6kAQa7bJ-O6HtnZ7XptBAPuwAx3FkLZpBJ7bdvuGAoCcF6iSSbakJYXKixLJ7ZGTVGfCaqrGDCPAyhFWBSEFlr6NMl6%2B1nWW3FkacKznHQxfJnnHGm3bTLwCwabfVmmFOlhbKJ4lW2YU6dugAnZA2-zGZJ2lF2ICAkCxGFJQAOQAGQMSsZR92V-q%2Bng4AYcVLUYa1Y2wV5NrzO4gPNBBTMyfBYPNGO49tUqHTkvxVHUL67mHKM0RjuWbVDCjdBhJMrTe0kQsLTiiJ4mts%2B5eCXhLuuVIrhCJwYOyZzV1qEqydgEI7%2BySOc-NQpLTuqxrKC2B7hsi4HsugX0qNBMs0ep3syfragMIiB0cgCvEhv5Pn8CN5spfXJXydsqbAve7kscB7325DPOeJZv7v25EFMagJvTEFosVSk756rPBaGUcQUhuDXELE3BAecAytDKC6cwbpqLVDKOYEgPgIhkPqiUKhx5jzVE-oRBe3cCFxR8Ew9uq9WGQRsigjiJhWycLQUtWAbJOHMIQN-ARHDqh6nkTAawdQlCPjAKKBAqAwDFGWGUGQYBrA0iAA&label=LYewrgzgpg%2Bg7lAlgcwBYBcYDMQCdhA)

### CODE REVIEW

File: lib/forms/CoBraLab-Mouse-Weight-Form/index.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Weight has to be in range for 0 to 45 grams'
+        })
+        return z.NEVER;
+      }
+
+      return parsed
+    }),
+    scaleUsed: z.string()
```
[the remark this answers is not in the document]
Comment (david-roper) [reply lib/forms/CoBraLab-Mouse-Weight-Form/index.ts:71]: This is due to how we use parseFloat whenever new numbers are types into the input, Josh is working on resolving this issue.

File: lib/forms/CoBraLab-Mouse-Weight-Form/index.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      kind: 'string',
+      variant: 'input',
+      label: 'Weight of mouse (in grams)'
+    },
+    scaleUsed: {
+      kind: 'string',
+      variant: 'input',
+      label: "Scale used"
+    }
+  },
```
[the remark this answers is not in the document]
Comment (david-roper) [reply lib/forms/CoBraLab-Mouse-Weight-Form/index.ts]: Some for of identification number/name

[the remark this answers is not in the document]
Comment (gdevenyi) [reply lib/forms/CoBraLab-Mouse-Weight-Form/index.ts]: Make that more explicit please. How do the students identify it?

[the remark this answers is not in the document]
Comment (david-roper) [reply lib/forms/CoBraLab-Mouse-Weight-Form/index.ts]: I will switch it serial code now, I can see if the also have a list of them which could make this a select field instead

File: lib/forms/CoBraLab-Mouse-Weight-Form/index.ts [hunk 3]
```diff
@@ -57,32 +57,8 @@ export default defineInstrument({
...
-          code: z.ZodIssueCode.custom,
-          message: 'Weight has to be in range for 0 to 45 grams'
-        })
-        return z.NEVER;
-      }
-
-      return parsed
-    }),
-    scaleID: z.string(),
+    mouseWeight: z.number().min(1).max(50),
```
Comment (gdevenyi) [inline lib/forms/CoBraLab-Mouse-Weight-Form/index.ts:60]: This is too small, our fat mouse project reached ~70g.

Move this limit to 100

Comment (david-roper) [reply lib/forms/CoBraLab-Mouse-Weight-Form/index.ts:60]: ah yes the high fat diet mice, i will increase it to consider them too

**Review by gdevenyi — APPROVED** [review]

**Pull request merged**